zhaquirks/zunzunbee/slateswitch.py

```diff
@@ -0,0 +1,150 @@
+                button = BUTTON_2
+                press_type = PRESS_TYPES[press]
+            elif value == 8:
+                button = BUTTON_3
+                press_type = PRESS_TYPES[press]
+            elif value == 16:
+                button = BUTTON_4
+                press_type = PRESS_TYPES[press]
+            elif value == 32:
+                button = BUTTON_5
+                press_type = PRESS_TYPES[press]
+            elif value == 64:
+                button = BUTTON_6
+                press_type = PRESS_TYPES[press]
+            elif value == 128:
+                button = BUTTON_7
+                press_type = PRESS_TYPES[press]
+            elif value == 256:
+                button = BUTTON_8
+                press_type = PRESS_TYPES[press]
+            else:
+                # discard invalid values:
+                return
+
+            action = f"{button}_{press_type}"
+            event_args = {
+                BUTTON: button,
+                PRESS_TYPE: press_type,
+            }
+            self.listener_event(ZHA_SEND_EVENT, action, event_args)
+
+
+class ZunZunBeeButton(CustomDevice):
+    """zunzunbee button device."""
+
+    signature = {
+        # <SimpleDescriptor endpoint=1 profile=260 device_type=1026
+        # device_version=0
+        # input_clusters=[0, 1, 3, 32, 1026, 1280, 2821]
+        # output_clusters=[3, 25]>
+        MODELS_INFO: [(ZUNZUNBEE, "SSWZ8T")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: 0x0104,
+                DEVICE_TYPE: 0x0104,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    Identify.cluster_id,
+                    PollControl.cluster_id,
+                    TemperatureMeasurement.cluster_id,
+                    IasZone.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Identify.cluster_id, Ota.cluster_id],
+            }
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    Identify.cluster_id,
+                    PollControl.cluster_id,
+                    TemperatureMeasurement.cluster_id,
+                    ZunZunBeeIASCluster,
+                ],
+                OUTPUT_CLUSTERS: [Identify.cluster_id, Ota.cluster_id],
+            }
+        }
+    }
+
+    device_automation_triggers = {
+        (press_type, button): {
+            COMMAND: f"{button}_{press_type}",
+            CLUSTER_ID: IasZone.cluster_id,
+        }
+        for press_type in (SHORT_PRESS, LONG_PRESS)
+        for button in (
+            BUTTON_1,
+            BUTTON_2,
+            BUTTON_3,
+            BUTTON_4,
+            BUTTON_5,
+            BUTTON_6,
+            BUTTON_7,
+            BUTTON_8,
+        )
```

Comment:
Note for me later: We need to add translations to ZHA for button 7 and 8: https://github.com/home-assistant/core/blob/982166df3cd420ea6b16c6b284838e0cea873a9f/homeassistant/components/zha/strings.json#L306-L311